transformer_engine/CMakeLists.txt

```diff
@@ -37,7 +37,3 @@ find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
 include_directories(${PROJECT_SOURCE_DIR})
 
 add_subdirectory(common)
-if(NVTE_WITH_USERBUFFERS)
```

Comment:
I think we can we get rid of this file altogether now and simply move all the above to `transformer_engine/common/CMakeLists.txt`?

Reply:
I agree, but I held off on it in case it ballooned into other changes in the build system and caused scope creep. I was planning to do a follow-up PR to remove this file and use `transformer_engine/common/CMakeLists.txt`, but if you think it's better to do it here, I can certainly add that.